### ISSUE TITLE
FURB188 doesn't raise if code has multiple conditions

### ISSUE THREAD

**Issue by gamer191** [astral-sh/ruff#26043]
### Summary

```
echo "s[len(a):] if s.startswith(a) else s"|ruff check --select FURB188 -
```
raises a [FURB188](https://docs.astral.sh/ruff/rules/slice-to-remove-prefix-or-suffix/#slice-to-remove-prefix-or-suffix-furb188) warning, however
```
echo "s[len(a):] if True and s.startswith(a) else s"|ruff check --select FURB188 -
```
does not

Whilst the latter can't be optimised as much, it can still be somewhat optimised
```diff
-s[len(a):] if True and s.startswith(a) else s
+s.removeprefix(a) if True else s
```

**Comment by ntBre** [comment]
It could be worth investigating how much this would complicate the rule. I think we could pretty much only handle this case, not other boolean operations and only this specific order of conditions, but it could still be reasonable to support if it doesn't take too much work. We may also want to check if refurb itself flags this case.

**Comment by gamer191** [comment]
> We may also want to check if refurb itself flags this case

I just checked and it doesn't. Should Ruff stay faithful to refurb's behaviour?

**Comment by ntBre** [comment]
I'd be curious to get others' thoughts on that. I'm not opposed to deviating from an upstream linter in general, as long as the change seems reasonable on its own and we can't find any explicit reason that the upstream tool made that decision. We wouldn't want to reopen a can of worms that they've previously fixed, for example.
